moveit_ros/moveit_servo/include/moveit_servo/servo_calcs.h

```diff
@@ -151,13 +151,16 @@ class ServoCalcs
    * Is handled differently for position vs. velocity control.
    */
   void suddenHalt(trajectory_msgs::msg::JointTrajectory& joint_trajectory) const;
-  void suddenHalt(sensor_msgs::msg::JointState& joint_state) const;
+  void suddenHalt(sensor_msgs::msg::JointState& joint_state,
+                  const std::vector<const moveit::core::JointModel*>& joints_to_halt) const;
 
   /** \brief  Scale the delta theta to match joint velocity/acceleration limits */
   void enforceVelLimits(Eigen::ArrayXd& delta_theta);
 
-  /** \brief Avoid overshooting joint limits */
-  bool enforcePositionLimits(sensor_msgs::msg::JointState& joint_state) const;
+  /** \brief Avoid overshooting joint limits
+      \return Vector of the joints that violate the position limits
```

Comment:
Should I apply the following suggestion .?

```suggestion
      \return Vector of the joints that are close to the position limits
```

Reply:
I think your revised wording is clearer.

Reply:
My suggestion would be

```suggestion
      \return Vector of the joints that increasingly violate the position margin limits
```

The comment in the first line of the enforcePositionLimits implementation is pretty descriptive, but probably too long for /brief
> // Halt if we're past a joint margin and joint velocity is moving even farther past